api/v1alpha1/nats_types.go

```diff
@@ -81,6 +64,82 @@ type NATS struct {
 	Status NATSStatus `json:"status,omitempty"`
 }
 
+// NATSStatus defines the observed state of NATS.
+type NATSStatus struct {
+	State      `json:"state"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// NATSSpec defines the desired state of NATS.
+type NATSSpec struct {
+	// Cluster defines configurations that are specific to NATS clusters.
+	Cluster `json:"cluster"`
+
+	// JetStream defines configurations that are specific to NATS JetStream.
+	// +optional
+	JetStream `json:"jetStream,omitempty"`
+
+	// JetStream defines configurations that are specific to NATS logging in NATS.
+	// +optional
+	Logging `json:"logging,omitempty"`
+
+	// Resources defines resources for NATS.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Annotations allows to add annotations to NATS.
+	// +optional
+	Annotations []string `json:"annotations,omitempty"`
+
+	// Labels allows to add Labels to NATS.
+	// +optional
+	Labels []string `json:"labels,omitempty"`
```

Comment:
Annotations and Labels needs to be `map[string]string`? dont they? They are key-value pairs

Reply:
https://github.com/kubernetes/kubernetes/blob/5f23f835879b514e9bf16809059b38957d20a486/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L223

you are right